src/main/kotlin/no/nav/yrkesskade/ysmeldingapi/controllers/handlers/RestResponseEntityHandler.kt

```diff
@@ -17,33 +18,44 @@ import javax.ws.rs.NotFoundException
 
 @ControllerAdvice
 class RestResponseEntityExceptionHandler : ResponseEntityExceptionHandler() {
+    private val log = getLogger(RestResponseEntityExceptionHandler::class.java)
+
     @ExceptionHandler(value = [BadRequestException::class])
     protected fun handleConflict(ex: RuntimeException?, request: WebRequest?): ResponseEntity<Any> =
-        handleExceptionInternal(ex!!, Feilmelding.fraException(ex), HttpHeaders(), HttpStatus.BAD_REQUEST, request!!)
+        handleExceptionAndLogError(ex, request, HttpHeaders(), HttpStatus.BAD_REQUEST)
 
     @ExceptionHandler(value = [NotFoundException::class])
     protected fun handleNotFound(ex: RuntimeException?, request: WebRequest?): ResponseEntity<Any> =
-        handleExceptionInternal(ex!!, Feilmelding.fraException(ex), HttpHeaders(), HttpStatus.NOT_FOUND, request!!)
+        handleExceptionAndLogError(ex, request, HttpHeaders(), HttpStatus.NOT_FOUND)
 
 
     @ExceptionHandler(value = [ForbiddenException::class])
     protected fun handleForbidden(ex: RuntimeException?, request: WebRequest?): ResponseEntity<Any> =
-        handleExceptionInternal(ex!!, Feilmelding.fraException(ex), HttpHeaders(), HttpStatus.FORBIDDEN, request!!)
+        handleExceptionAndLogError(ex, request, HttpHeaders(), HttpStatus.FORBIDDEN)
 
     @ExceptionHandler(value = [AltinnException::class])
     protected fun handleAltinnException(ex: AltinnException?, request: WebRequest?): ResponseEntity<Any> =
-        handleExceptionInternal(ex!!, Feilmelding.fraException(ex), HttpHeaders(), HttpStatus.valueOf(ex.httpStatus), request!!)
+        handleExceptionAndLogError(ex, request, HttpHeaders(), HttpStatus.valueOf(ex!!.httpStatus))
 
     @ExceptionHandler(value = [InvalidFormatException::class])
     protected fun handleJacksonSerializationExceptions(ex: Exception?, request: WebRequest?): ResponseEntity<Any> =
-        handleExceptionInternal(ex!!, Feilmelding.fraException(ex), HttpHeaders(), HttpStatus.BAD_REQUEST, request!!)
+        handleExceptionAndLogError(ex, request, HttpHeaders(), HttpStatus.BAD_REQUEST)
+
+    @ExceptionHandler(value = [Exception::class])
+    protected fun handleAnyException(ex: Exception?, request: WebRequest?): ResponseEntity<Any> =
+        handleExceptionAndLogError(ex, request, HttpHeaders(), HttpStatus.BAD_REQUEST)
+
+    private fun handleExceptionAndLogError(ex: Exception?, request: WebRequest?, headers: HttpHeaders, httpStatus: HttpStatus): ResponseEntity<Any> {
+        log.error("${ex!!.javaClass.simpleName}: ${ex!!.message} \n${ex!!.stackTraceToString()}")
```

Comment:
Er det trygt med !! når signaturen på funksjonen sier at parametrene kan være null? 
Og dersom vi vet at params ikke er null, burde det parametrene ikke vært definert som nullable? Antar det er for å ett sted nullable og !!, i stedet for !! ved kall fra alle de andre funksjonene...

Reply:
Jeg sjekket handleExceptionInternal og ser at den ikke har nullable parametere, så jeg fjernet også det alternativet fra denne metoden.